src/runtime/threadpool/builder.rs

```diff
@@ -172,12 +186,19 @@ impl Builder {
             .enable_all()
             .on_thread_start(move || {
                 // We need the threadpool's sender to set up the default tokio
-                // 0.1 executor.
-                let lock = compat_sender2.read().unwrap();
+                // 0.1 executor. We also need to upgrade the weak pointer, if the
+                // pointer is no longer valid then the runtime has shutdown and the
+                // handle is no longer available.
+                //
+                // This upgrade will only fail if the compat runtime has been dropped.
+                let sender = compat_sender2.upgrade().expect("Compat runtime shutdown.");
+                let lock = sender.read().unwrap();
                 let compat_sender = lock
                     .as_ref()
                     .expect("compat executor needs to be set before the pool is run!")
                     .clone();
+                drop(lock);
+                drop(sender);
```

Comment:
these should be dropped after the call to `set_guards` anyway, as the function returns — is there a reason we need to drop them _before_ calling `set_guards`? if so, it would be nice to have a comment explaining that.

Reply:
Yeah, I mostly added them to be explicit, will remove them :+1: